lib/internal/util/inspect.js

```diff
@@ -1026,6 +1031,11 @@ function formatRaw(ctx, value, recurseTimes, typedArray) {
       if (keys.length === 0 && protoProps === undefined) {
         return base;
       }
+    } else if (isURL(value) && !(recurseTimes > ctx.depth && ctx.depth !== null)) {
```

Comment:
```suggestion
    } else if (typeof value.href === 'string' && value instanceof URL) {
```

If this is used for now, I'd approve the PR. This limits the detection to definite URLs while it won't be able to print all cases.